File: src/dotnet/Src/ButterflyNet.Parser/Scopes/ListItemScope.cs
namespace ButterflyNet.Parser.Scopes {
	public class ListItemScope : BlockScope {
		public ListItemScope(int depth) {
			Depth = depth;
		}

		public int Depth { get; private set; }

		public override void Open(ButterflyAnalyzer analyzer) {
			analyzer.OpenListItem();
		}

		public override void Close(ButterflyAnalyzer analyzer) {
			analyzer.CloseListItem();
		}
	}
}

File: src/js/src/scopes/ListItemScope.js
function ListItemScope(depth) {
	this.depth = depth;
	this.type = ScopeTypeCache.listItem;
	this.level = ScopeLevel.block;
	this.canNestText = true;
	this.open = function(analyzer) {
		analyzer.openListItem();
	};
	this.close = function(analyzer) {
		analyzer.closeListItem();
	};
}

extend(Scope, ListItemScope);